### ISSUE TITLE
capitalized-comments' ignorePattern should default to no-fallthrough's commentPattern

### ISSUE THREAD

**Issue by revolter** [eslint/eslint#10662]
**Tell us about your environment**

* **ESLint Version:** 5.2.0
* **Node Version:** v8.9.4
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser": true,
        "es6": true,
        "greasemonkey": true
    },
    "extends": "eslint:recommended",
    "globals": {
        "player": false,
        "PLAYER_STATE": false
    },
    "rules": {
        "no-fallthrough": [
            "error", {
                "commentPattern": "fall\\-through"
            }
        ],
        "capitalized-comments": "error"
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var playerState = player.getPlayerState();

switch (playerState) {
    case PLAYER_STATE.Unstarted: // fall-through
    case PLAYER_STATE.Ended: // fall-through
    case PLAYER_STATE.Paused: // fall-through
    case PLAYER_STATE.Buffering: // fall-through
    case PLAYER_STATE.VideoCued: {
        player.playVideo();

        break;
    }
}
```

```bash
eslint scripts/**/*
```

**What did you expect to happen?**

`capitalized-comments` should have been loaded the `ignorePattern` (that I didn't set) from the `no-fallthrough`'s `commentPattern` option, as it's clear I'd be using that fall through pattern.

**What actually happened? Please include the actual, raw output from ESLint.**

```
.../scripts/Main/main.js
  4:34  error  Comments should not begin with a lowercase character  capitalized-comments
  5:30  error  Comments should not begin with a lowercase character  capitalized-comments
  6:31  error  Comments should not begin with a lowercase character  capitalized-comments
  7:34  error  Comments should not begin with a lowercase character  capitalized-comments
```

**Comment by kaicataldo** [comment]
Hi, thanks for the issue. ESLint rules can't know about each other by design, so this is not something `capitalized-comments` can do. A solution for this is to use the `capitalized-comments` rule's `ignorePattern` option:
```
/* eslint capitalized-comments: ["error", "always", { "ignorePattern": "fall-through" }] */
```
I don't think adding a case by default for the `no-fallthrough` is desirable here for two reasons:
 1. The pattern can be changed
 1. If someone isn't using `no-fallthrough`, the `capitalized-comments` rule could be ignoring comments it shouldn't be

**Comment by kaicataldo** [comment]
> I meant adding a default to the `capitalized-comments`, not `no-fallthrough`.

Sorry, looks like I had a typo that made that unclear - fixed in the comment above.

If `capitalized-comments` had a default value for the `no-fallthrough` comment, it would be incorrect if the `no-fallthrough` comment pattern was changed (as it is in your case). i.e. the default would be scanning for `/falls?\s?through/i` (the default in `no-fallthrough` and would miss your comments because you changed `commentPattern` in your config.

**Comment by revolter** [comment]
I meant for `capitalized-comments` rule's `ignorePattern` option to have a default value equal to the `no-fallthrough` rule's `commentPattern` option.

Here are some examples:

1.

```js
{
    "rules": {
        "capitalized-comments": "error"
    }
}
```

`capitalized-comments` rule's `ignorePattern` option's value will be `null` like in present. This is because `no-fallthrough` is not enabled.

2.

```js
{
    "rules": {
        "no-fallthrough": "error",
        "capitalized-comments": "error"
    }
}
```

<strike>`capitalized-comments` rule's `ignorePattern` option's value will be `falls?\s?through` instead of `null`. This is because `no-fallthrough` is enabled, and its `commentPattern` option defaults to `falls?\s?through`.</strike>
`capitalized-comments` rule's `ignorePattern` option's value will be `null` like in present. This is because `no-fallthrough` is enabled, and its `commentPattern` option defaults to `/falls?\s?through/i` which is case insensitive.

3.

```js
{
    "rules": {
        "no-fallthrough": [
            "error", {
                "commentPattern": "my-custom-string"
            }
        ],
        "capitalized-comments": "error"
    }
}
```

`capitalized-comments` rule's `ignorePattern` option's value will be `my-custom-string` instead of `null`. This is because `no-fallthrough` is enabled, and its `commentPattern` option is set to `my-custom-string` which is case sensitive.

4.

```js
{
    "rules": {
        "no-fallthrough": [
            "error", {
                "commentPattern": "my-custom-string"
            }
        ],
        "capitalized-comments": [
            "error", {
                "ignorePattern": "my-other-custom-string"
            }
        ]
    }
}
```

`capitalized-comments` rule's `ignorePattern` option's value will be `my-other-custom-string` instead of `null` like in present. This is because even though `no-fallthrough` is enabled and its `commentPattern` option is set to `my-custom-string`, the `capitalized-comments` rule's `ignorePattern` option's value is set to `my-other-custom-string`.

**Comment by kaicataldo** [comment]
I understand, but as mentioned above, ESLint rules can't know about each other (and as an extension, their configs) by design. I definitely understand the intent of this issue - it would be a nice DX change to not have to configure `capitalized-comments` yourself.

**Comment by revolter** [comment]
And is this (rules knowing about other rules) something you guys would never want to have in ESLint?

**Comment by not-an-aardvark** [comment]
"Never" is a strong word, but I'm fairly certain this isn't something we would want right now. By design, rules are completely independent of each other, and this has several useful properties (e.g. rules don't have to worry about what other rules are active or available, and users have a lot of flexibility to configure rules without worrying about whether the configuration of one rule will affect any of the other rules they have enabled).

**Comment by revolter** [comment]
I didn't know how to formulate it better, though I know it wasn't the best (not a native english speaker). So I guess this can be closed now, right?

Fun fact, I currently have this:

```js
{
    "rules": {
        "no-fallthrough": [
            "error", {
                "commentPattern": "fall\\-through"
            }
        ],
        "capitalized-comments": [
            "error",
            "always", {
                "ignorePattern": "fall\\-through"
            }
        ],
        "line-comment-position": [
            "error", {
                "position": "above",
                "ignorePattern": "fall\\-through"
            }
        ]
    }
}
```

**Comment by not-an-aardvark** [comment]
No worries -- I just meant that it's hard for me to predict how ESLint will change in the distant future.

Yes, I'll go ahead and close it. We definitely appreciate you creating the issue!

To avoid duplication in your config, one option would be to use an `.eslintrc.js` file and use variables:

```js
const FALL_THROUGH_PATTERN = "fall\\-through";

module.exports = {
    "rules": {
        "no-fallthrough": [
            "error", {
                "commentPattern": FALL_THROUGH_PATTERN
            }
        ],
        "capitalized-comments": [
            "error",
            "always", {
                "ignorePattern": FALL_THROUGH_PATTERN
            }
        ],
        "line-comment-position": [
            "error", {
                "position": "above",
                "ignorePattern": FALL_THROUGH_PATTERN
            }
        ]
    }
};
```

**Comment by revolter** [comment]
Very good idea 👍 Thank you!
